### PR TITLE
Ddt 878 get shipment events itests (EBL)

### PR DESCRIPTION
**Pull request by FinnAtAsseco**

### CODE REVIEW

File: src/test/java/ebl/v2/ShipmentEventIT.java [hunk 1]
```diff
@@ -27,14 +33,18 @@ static void configs() throws IOException {
...
       .contentType("application/json")
       .get("/v2/events")
       .then()
       .assertThat()
       .statusCode(200)
       .contentType(ContentType.JSON)
+      .header("API-Version", equalTo("2.0.0"))
+      .header("Current-Page", matchesRegex("^http://.*/v2/events\\?cursor=[a-zA-Z\\d]*$"))
+      .header("Next-Page", matchesRegex("^http://.*/v2/events\\?cursor=[a-zA-Z\\d]*$"))
+      .header("Last-Page", matchesRegex("^http://.*/v2/events\\?cursor=[a-zA-Z\\d]*$"))
```
Comment (nt-as) [inline src/test/java/ebl/v2/ShipmentEventIT.java:47]: I would prefer `https?` (I know it is unlikely to matter for the IT for eBL, but the pattern will likely be copy-wasted many places, so it seems like a small price to future proof it)

Comment (FinnAtAsseco) [reply src/test/java/ebl/v2/ShipmentEventIT.java:47]: Good point

File: src/test/java/ebl/v2/ShipmentEventIT.java [hunk 2]
```diff
@@ -116,8 +126,8 @@ void testGetAllEventsByCarrierBookingReference() {
       .body("eventType", everyItem(equalTo("SHIPMENT")))
       .body("eventClassifierCode", everyItem(equalTo("ACT")))
       .body("documentTypeCode", everyItem(anyOf(equalTo("SHI"), equalTo("TRD"))))
-    // TODO: Assert that document references contain the carrier booking reference 832deb4bd4ea4b728430b857c59bd057
-    // (Needs a fix for DDT-928 first)
+      .body("documentReferences.flatten().findAll { it.documentReferenceType == 'BKG' }.size()", greaterThanOrEqualTo(3))
+      .body("documentReferences.flatten().findAll { it.documentReferenceType == 'BKG' }.documentReferenceValue", everyItem(equalTo("832deb4bd4ea4b728430b857c59bd057")))
```
Comment (pimg) [inline src/test/java/ebl/v2/ShipmentEventIT.java:130]: missing json schema validation

File: src/test/java/ebl/v2/ShipmentEventIT.java [hunk 3]
```diff
@@ -162,12 +193,14 @@ void testGetAllEventsByCarrierBookingReferenceWithEventCreatedDateTimeRange() {
...
+      .body("eventCreatedDateTime", everyItem(
+        asDateTime(
+          allOf(
             greaterThanOrEqualTo(ZonedDateTime.parse(rangeStart)),
             lessThan(ZonedDateTime.parse(rangeEnd))
       ))))
-      // TODO: Assert that document references contain the carrier booking reference 832deb4bd4ea4b728430b857c59bd057
-      // (Needs a fix for DDT-928 first)
+      .body("documentReferences.flatten().findAll { it.documentReferenceType == 'BKG' }.size()", greaterThanOrEqualTo(3))
+      .body("documentReferences.flatten().findAll { it.documentReferenceType == 'BKG' }.documentReferenceValue", everyItem(equalTo("832deb4bd4ea4b728430b857c59bd057")))
```
Comment (pimg) [inline src/test/java/ebl/v2/ShipmentEventIT.java:203]: missing json schema validation

**Review by pimg — CHANGES_REQUESTED** [review]

File: src/test/java/ebl/v2/ShipmentEventIT.java (hunk 2 above)
Comment (FinnAtAsseco) [reply src/test/java/ebl/v2/ShipmentEventIT.java:130]: The alleventsand headers test already have
.body(jsonSchemaValidator("shipmentEvent"));

Should we put schemavalidations in every test?

**Review by pimg — APPROVED** [review]

**Comment by douradofrazer** [comment]
A rebase with the current master will fix the failing CI.

**Review by pimg — APPROVED** [review]

**Pull request merged**